synapse/handlers/oidc_handler.py

```diff
@@ -93,6 +94,9 @@ class OidcHandler(BaseHandler):
     """Handles requests related to the OpenID Connect login flow.
     """
 
+    # The number of attempts to ask the mapping provider for when generating an MXID.
+    _MAP_USERNAME_RETRIES = 1000
```

Comment:
Hmm, this seems a bit excessive and like we're trying to sweep the problem under the carpet. I can foresee people just increment an integer on the end, which would end up failing for the 1000th user.

Not sure what to do instead though :/

Reply:
This is what we do for SAML currently, but it did seem rather excessive to me too.

Reply:
I guess that's fine for now then.

Reply:
See #8813 also.